<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" dir="ltr">

<head>
	<meta http-equiv="content-type" content="text/html; charset=UTF-8"/>
	<meta name="description" content=""/>
	<meta name="keywords" content="" />
	<meta name="author" content="" />
	<title>ECCB 2014 T01</title>
	<link rel="stylesheet" type="text/css" href="style.css" media="screen" />
<style type="text/css">

#wrapper{
  margin:0 auto;
  #padding:15px 15% 8em;
  text-align:left
}
#content {
  max-width:70em;
  width:100%;
  margin:0 auto;
  padding-bottom:20px;
  overflow:hidden
}
.demo {
  margin:1.5em 0;
  padding:1.5em 1.5em 0.75em;
  border:1px solid #ccc;
  position:relative;
  overflow:hidden
}

.post {
  position:relative;
  overflow:hidden
}
.collapse p {padding:0 10px 1em}

.switch {position:absolute; top:1.5em; right: 1.5em; padding:3px}

#.post .switch {position:static; text-align:right}

.post .main{margin-bottom:0; padding-bottom:0}

.other li, .summary {margin-bottom:.3em; padding:1em; border:1px solid #e8e7e8; background-color:#f8f7f8}

.other ul {list-style-type:none; text-align:center}


.expand{padding-bottom:.75em}

/* --- Links  --- */

#download {
  border-style:none;
  background:white;
}

.expand a {
  display:block;
  padding:3px 10px
}
.expand a:link, .expand a:visited {
  border-width:1px;
  background-image:url(img/arrow-down.gif);
  background-repeat:no-repeat;
  background-position:98% 50%;
}
.expand a:hover, .expand a:active, .expand a:focus {
}
.expand a.open:link, .expand a.open:visited {
  #border-style:solid;
  #background:#eee url(img/arrow-up.gif) no-repeat 98% 50%
  color:black;
}
</style>
<!--[if lte IE 6]>
<style type="text/css">
h3 a, .demo {position:relative; height:1%}
</style>
<![endif]-->

<!--[if lte IE 6]>
<script type="text/javascript">
   try { document.execCommand( "BackgroundImageCache", false, true); } catch(e) {};
</script>
<![endif]-->
<!--[if !lt IE 6]><!-->
<script type="text/javascript" src="http://ajax.googleapis.com/ajax/libs/jquery/1.4.2/jquery.min.js"></script>
<script type="text/javascript" src="scripts/expand.js"></script>
<script type="text/javascript">
<!--//--><![CDATA[//><!--
$(function() {
    $("#content h3.expand").toggler();
    $("#content h2.expand").toggler();
    $("#content div.demo").expandAll({trigger: "h3.expand", ref: "h3.expand"});
    $("#content div.other").expandAll({
      expTxt : "[Show]", 
      cllpsTxt : "[Hide]",
      ref : "ul.collapse",
      showMethod : "show",
      hideMethod : "hide"
    });
    $("#content div.post").expandAll({
      expTxt : "[Show tip]", 
      cllpsTxt : "[Hide tip]",
      ref : "div.collapse", 
      localLinks: "p.top a"    
    });    
});
//--><!]]>
</script>
<!--<![endif]-->


    <script type="text/javascript" src="syntaxhighlight/shCore.js"></script>
    <script type="text/javascript" src="syntaxhighlight/shBrushBash.js"></script>
    <script type="text/javascript" src="syntaxhighlight/shBrushR.js"></script>
    <link type="text/css" rel="stylesheet" href="syntaxhighlight/shCore.css"/>
    <link type="text/css" rel="stylesheet" href="syntaxhighlight/shThemeDefault.css"/>
    
    <script type="text/javascript">
      SyntaxHighlighter.config.clipboardSwf = 'syntaxhighlight/clipboard.swf';
      SyntaxHighlighter.all();
      
    </script>

</head>

<body>

<div id="site-wrapper">

	<div id="header">

		<div id="top">

			<div class="left" id="logo">
				<a href="#"><h2 class="label label-green">Session 1 : Website</h2></a>
			</div>



			<div class="clearer">&nbsp;</div>

		</div>

		<div class="navigation" id="sub-nav">

			<ul class="tabbed">
				<li ><a href="index.html">Home</a></li>
				<li ><a href="session1.html">Session1:Web site</a></li>
				<li><a href="session2.html">Session2:Command line</a></li>
				<li><a class="current-tab" href="session3.html">Session3:SOAP Web services</a></li>
			</ul>

			<div class="clearer">&nbsp;</div>

		</div>

	</div>

	<div class="main" id="main-two-columns">

		<div class="left" id="main-content">

			<h2 id="contents"> Introduction </h2>
			<b>Goal</b><br/>

<u>The aim is to :</u>
<ul>
<li> get familiar with remote programmatic access using SOAP Web services</li>
<li> use client script in Python to query RSAT Web Services (WS)</li>
</ul>
<u>In practice :</u>
<ul>
<li> extract sequences corresponding to peaks (<i>fetch-sequences</i>)</li>
<li>scan the peak sequences with a discovered motif (<i>matrix-scan</i>)</li>
</ul>
						
	<div id="wrapper"> 
      <div id="content"> 
      
       <!-- INSTALL Suds -->	 
          <div class="demo">
            <h2 id="download" class="expand">Install the Python modules for SOAP</h2>
            <div class="collapse">
            <div class="notice">
			<b>Goal:</b> prepare the Python environment to access remote SOAPweb services
			<br/>
			</div>
            <h3 class="expand">1 - On MAC/Linux </h3>
            <div class="collapse">
            We will first install <a href="http://pip.readthedocs.org/en/latest/installing.html#using-package-managers" target="_blank"> pip as follow </a>:
            <pre>sudo easy_install pip</pre>
            Alternative : Linux users may prefer 
            <pre> apt-get install python-pip </pre>
            
            <br/>
            Now we will install <a href="https://pypi.python.org/pypi/suds" target="_blank"> suds</a>  via pip:
			<pre>sudo pip install suds</pre>
			 Alternative :Linux users may prefer 
            <pre>apt-get install python-suds</pre>
			
            </div>
            
		<h3 class="expand">2 - On Windows </h3>
		<div class="collapse">
                  <ol>
		    <li>Install pip with <a href="http://pip.readthedocs.org/en/latest/installing.html#using-package-managers" target="_blank"> these instructions </a>. Note from <a href="http://stackoverflow.com/questions/4750806/how-to-install-pip-on-windows" target="_blank">Stackoverflow</a>: pip.exe will be placed inside your Python installation's Scripts folder, which is likely not on your path (fix that by running C:\PythonXX\Tools\Scripts\win_add2path.py. Download get-pip.py, being careful to save it as a .py file rather than .txt.</li>
		
            <li>Now we will install <a href="https://pypi.python.org/pypi/suds" target="_blank"> suds</a>  via pip:
			<pre> pip install suds</pre>
		  </li>
	    
	    </div>
            <div class="error">At this point, you're ready to access remote SOAP webservices, including the RSAT WS. </div>
			
            </div>         
	    
            </div>         
      
      <!-- SEQUENCES -->
                <div class="demo">
            <h2 id="download" class="expand">Retrieving sequences from your peaks</h2>
            <div class="collapse">
            <div class="notice">
			<b>Goal:</b> Given a set of peaks from a ChIP-seq experiment in a bed format, retrieve the sequences corresponding to those coordinates from the genome in fasta format.  
			<br/>
			</div>
            <h3 class="expand">1 - Example dataset1: CEBPa binding regions in dog liver </h3>
            <div class="collapse">
          <p>Schmidt, Wilson and Ballester published a ChIP-seq experiment on liver tissue to identify binding regions for the transcription factor CEBPa <a href="http://www.sciencemag.org/content/328/5981/1036" target='_blank'>(PMID:20378774)</a> in five different species (human, mouse, dog, short-tailed opossum and chicken). This data set is publicly available through arrayexpess<a href="http://www.ebi.ac.uk/arrayexpress/experiments/E-TABM-722/" target='_blank'> (E-TABM-722)</a>.</br>
	  As done by the authors, CEBPa binding regions (peaks) were called using SWEMBL with parameter R=0.05, merged reads from two biological replicates and their corresponding input controls. For this tutorial we will analyze CEBPa binding pattern in dog, peaks can be downloaded from <a href="./data/dataset_cebpa_dog/do61+do79_cfam_CEBPA_liver.bed.SWEMBL.3.3.bed">here</a>.<br/> 	
            </div>
            <h3 class="expand">2 - Fetch sequences from a bed file  </h3>
            <div class="collapse">
 			<ol>
                <li>Save this Python script <a href="python_scripts/fetch-sequences_soap.py">fetch-sequences_soap.py</a> on your computer</li>
                <li>Run the script a first time to get the help by typing
                <pre>python fetch-sequences_soap.py -h</pre>
                <br/>
                 
                <div class="post">
   				<div class="success"> Which files do you need, according to the help of the program ?<br>
                </div>
            	<p class="switch"><a href="#" class="">[Show tip]</a></p><div style="display: none;" class="collapse"> 
                <pre>usage: fetch-sequences_soap.py  -b BED_FILE -g GENOME_NAME</pre>
                           You need a coordinate file (peaks) in BED format.
                </div>
                 </li>
                <li> Now run the script to retrieve the sequences corresponding to the <a href="data/dataset_cebpa_dog/do61+do79_cfam_CEBPA_liver.bed.SWEMBL.3.3.bed">CEBPa peak coordinates</a>. Save this file on your machine and adapt the file path of the command to the files as needed for your computer. The genome is the dog CanFam2 assembly</li>
                
                <pre>python fetch-sequences_soap.py -b do61+do79_cfam_CEBPA_liver.bed.SWEMBL.3.3.bed -g canFam2</pre>
                <br/>
                <li> You should obtain the results from fetch-sequences. Note that this analysis did not run on your computer, but used the computing power of the remote RSAT server.</li>
      
                </ol> 

                         </div>
            
            <div class="error">At this point, you're able to query an RSAT WS, and programmatically extract sequences corresponding to a given set of genomic coordinates. </div>
			
            </div>         
	    
            </div>  
               
          
     <!-- MATRIX-SCAN-->	     
          
          
          <div class="demo">
            <h2 id="quality" class="expand" >Scanning peak sequences with a discovered motif</h2>
            <div class="collapse">
            <div class="notice">
			<b>Goal:</b> Given peak sequences in FASTA format and a motif represented by a matrix in TRANSFAC format, obtain the location of putative TFBS.
			<br/>
			</div>
            <h3 class="expand">1 - Run matrix-scan as web service</h3>
            <div class="collapse">
                <ol>
                <li>Save this Python script <a href="python_scripts/matrix_scan_soap.py">matrix_scan_soap.py</a> on your computer</li>
                <li>Run the script a first time to get the help by typing
                <pre>python matrix_scan_soap.py -h</pre>
                <br/>
                 
                <div class="post">
   				<div class="success"> Which files do you need, according to the help of the program ?<br>
                </div>
            	<p class="switch"><a href="#" class="">[Show tip]</a></p><div style="display: none;" class="collapse"> 
                <pre>usage: matrix_scan_soap.py [-h] -s SEQUENCE_FILE -m MATRIX_FILE -u P-VALUE</pre>
                           You need a sequence file in FASTA format and a matrix file in TRANSFAC format.
                </div>
                 </li>
                <li> Now run the script to scan the <a href="results/session1/1_do61+do79_cfam_CEBPA_liver.SWEMBL.3.3_peaks.fasta">CEBPa peak sequences</a> with the first discovered motif (session1, <a href="data/session3/peak-motifs_oligos_6nt_mkv4_m1.tf">oligos_6nt_mkv4_m1</a>). Adapt the file path to the files as needed for your computer.</li>
                
                <pre>python matrix_scan_soap.py -s 1_do61+do79_cfam_CEBPA_liver.SWEMBL.3.3_peaks.fasta -m peak-motifs_oligos_6nt_mkv4_m1.tf -u 0.0001 > cebpa_oligos_6nt_mkv4_m1_10-4.ft</pre>
               </br>
                <li> You should obtain the results from matrix-scan within the file cebpa_oligos_6nt_mkv4_m1_10-4.ft. Note that this analysis did not run on your computer, but used the computing power of the remote RSAT server.</li>
      
                </ol> 
            </div>

            <div class="error">At this point, you should know what is a Web service and be able to run a pre-made client script in Python.
                </div>
          </div>
          </div>
        
                                
          </div>
          
	    
	    

	    	    </div>	
	    	    	</div>
	    	    	</div>
	    	    	
	
<!-- SIDE BAR -->		

		<div class="right sidebar" id="sidebar">

			<h5>Slides:</h5>
			<ul>
			<li><a href="booklet/booklet_chip-seq.pdf" target='_blank'>Presentation</a></li>
			</ul>
			<h5>Dataset:</h5>
			Schmidt,Wilson and Ballester <i>Five-vertebrate ChIP-seq reveals the evolutionary dynamics of transcription factor binding.</i> Science, 328(5981), 1036â€“1040. 2010 <a href="http://www.sciencemag.org/content/328/5981/1036" target='_blank'>Article</a>
			<a href="articles/2013_Myers_PLoS Genet.pdf" target='_blank'> [PDF] </a><br/>
			<u>peaks:</u><a href="peaks_url.bed" target='_blank'> bed file</a><br/>		
			</ul>
			
			<p/>
			<h5>RSAT WSDL:</h5>
			<ul>
			<li><a href ="http://rsat.ulb.ac.be/rsat/web_services/RSATWS_documentation.xml" target='_blank'>human-readable</a></li>
			<li><a href ="http://rsat.ulb.ac.be/rsat/web_services/RSATWS.wsdl" target='_blank'>WSDL</a></li>
			</ul>
			
					
			<h5>Suggested Reading:</h5>
			<ul>
			<li>Sand O, Thomas-Chollier M, Vervisch E, van Helden J.
<i>Analyzing multiple datasets by interconnecting RSAT programs via SOAP Web Services - an example with ChIP-chip data</i>,
Nature Protocols 3(10), 1604-15 (2008) <a href="http://www.ncbi.nlm.nih.gov/pubmed/18802441?ordinalpos=1&itool=EntrezSystem2.PEntrez.Pubmed.Pubmed_ResultsPanel.Pubmed_DiscoveryPanel.Pubmed_Discovery_RA&linkpos=3&log$=relatedarticles&logdbfrom=pubmed" target='_blank'> [Pubmed] </a><br/></li>	 </ol>
			</li>

			

		</div>

		<div class="clearer">&nbsp;</div>

	</div>

	<div id="footer">

		<div class="left" id="footer-left">
			
		
			
			<p>&copy; 2014 Morgane Thomas-Chollier. All rights Reserved</p>

			<p class="quiet"><a href="http://templates.arcsin.se/">Website template</a> by <a href="http://arcsin.se/">Arcsin</a></p>
			
			<div class="clearer">&nbsp;</div>

		</div>

		<div class="right" id="footer-right">

		</div>

		<div class="clearer">&nbsp;</div>

	</div>

</div>

</body>
</html>